tests/unit/test_locationHelper.py

```diff
@@ -0,0 +1,191 @@
+		self.assertIn(Point(x=4, y=4), rect)
+		self.assertNotIn(Point(x=2, y=6), rect)
+		self.assertNotIn(Point(x=6, y=2), rect)
+		self.assertNotIn(Point(x=6, y=6), rect)
+
+	def test_equal(self):
+		self.assertEqual(RectLTRB(left=2, top=2, right=4, bottom=4), RectLTRB(left=2, top=2, right=4, bottom=4))
+		self.assertNotEqual(RectLTRB(left=2, top=2, right=4, bottom=4), RectLTRB(left=2, top=2, right=6, bottom=6))
+		self.assertEqual(RectLTRB(left=2, top=2, right=4, bottom=4), RectLTWH(left=2, top=2, width=2, height=2))
+		self.assertNotEqual(RectLTRB(left=2, top=2, right=4, bottom=4), RectLTWH(left=2, top=2, width=4, height=4))
+
+	def test_ctypesRECT(self):
+		# Intersection
+		self.assertEqual(RectLTRB(left=2, top=2, right=4, bottom=4).intersection(RECT(left=3, top=3, right=5, bottom=5)), RectLTRB(left=3, top=3, right=4, bottom=4))
+		# Superset
+		self.assertTrue(RectLTRB(left=2, top=2, right=6, bottom=6).isSuperset(RECT(left=2, top=2, right=4, bottom=4)))
+		# Subset
+		self.assertTrue(RectLTRB(left=2, top=2, right=4, bottom=4).isSubset(RECT(left=2, top=2, right=6, bottom=6)))
+		# in
+		self.assertIn(RECT(left=2, top=2, right=4, bottom=4), RectLTRB(left=2, top=2, right=6, bottom=6))
+		self.assertNotIn(RECT(left=2, top=2, right=4, bottom=4), RectLTRB(left=2, top=2, right=4, bottom=4))
+		# Equality
+		self.assertEqual(RECT(left=2, top=2, right=4, bottom=4), RectLTRB(left=2, top=2, right=4, bottom=4))
+		self.assertNotEqual(RECT(left=2, top=2, right=4, bottom=4), RectLTRB(left=2, top=2, right=6, bottom=6))
+
+class TestRectUtilities(unittest.TestCase):
+
+	def test_points(self):
+		rect = RectLTRB(left=-5, top=-5, right=5, bottom=5)
+		self.assertEqual(rect.topLeft, Point(x=-5, y=-5))
+		self.assertEqual(rect.topRight, Point(x=5, y=-5))
+		self.assertEqual(rect.bottomLeft, Point(x=-5, y=5))
+		self.assertEqual(rect.bottomRight, Point(x=5, y=5))
+		self.assertEqual(rect.center, Point(x=0, y=0))
+		# Specifically test some other edge cases for center
+		self.assertEqual(RectLTRB(left=10, top=10, right=20, bottom=20).center, Point(x=15, y=15))
+		self.assertEqual(RectLTRB(left=-20, top=-20, right=-10, bottom=-10).center, Point(x=-15, y=-15))
+		self.assertEqual(RectLTRB(left=10, top=10, right=21, bottom=21).center, Point(x=16, y=16))
+		self.assertEqual(RectLTRB(left=-21, top=-21, right=-10, bottom=-10).center, Point(x=-16, y=-16))
+
+class TestToRectLTRB(unittest.TestCase):
+
+	def test_collection(self):
+		"""Tests whether a collection of several rectangle and point types convert to the expected L{RectLTRB}."""
+		rect=RectLTRB(left=10, top=15, right=500, bottom=1000)
+		self.assertEqual(toRectLTRB(
+			rect.topLeft, 
+			rect.bottomRight, 
+			rect.center, 
+			Point(15, 15),
+			Point(20, 20),
+			Point(50, 50),
+			Point(400, 400),
+			POINT(x=15, y=15),
+			POINT(x=20, y=20),
+			POINT(x=50, y=50),
+			POINT(x=400, y=400),
+			RectLTRB(left=450, top=450, right=490, bottom=990),
+			RECT(450, 450, 490, 990)
+		), rect)

```

Comment:
toRectLTRB and toRectLTWH always return only one rectangle containing all points in the parameters. For example:

> `locationHelper.toRectLTRB(locationHelper.Point(3,3), locationHelper.Point(4,4))`
> `RectLTRB(left=3, top=3, right=4, bottom=4)`

I realise now that there are still some inconsistencies with regard to inclusivity and exclusivity. toRectLTRB(Point(3,3)) returns RectLTRB(3,3,4,4), and toRectLTRB(Point(3,3), Point(4,4)) returns the same. The reason for this is that toRectLTRB converts a single point to a rectangle with a width and height of 1.

Reply:
I wondered if it might do something like that. Is this actually used anywhere yet? If not, I suggest we remove it. If it is, then I would prefer a dedicated function with a name that conveys this better eg `toRectLTRBContainingAll` and has its own documentation.

> I realise now that there are still some inconsistencies with regard to inclusivity and exclusivity. 
But it sounds like this also needs to be fixed. Perhaps some more tests for this as well.

Reply:
By the way, there are probably some good names for functions like this in collision detection libraries used in games. You would be looking for something that "creates the minimum volume axis-aligned 2D bounding box".

Reply:
To make sure, are you suggesting the following way of splitting the functionality of toRectLTRB and TORectLTWH?

* Make toRectLTRB and toRectLTWH only do conversion from rect classes and tuples of four integers
* Create separate functions to create bounding rectangles from several types of input